redis/src/cluster_async/mod.rs

```diff
@@ -675,15 +714,37 @@ where
             inner.cluster_params.tls,
             num_of_nodes_to_query,
         )?;
-
+        // Create a new connection vector of the found nodes
         let connections: &ConnectionMap<C> = &read_guard.0;
         let mut nodes = new_slots.values().flatten().collect::<Vec<_>>();
         nodes.sort_unstable();
         nodes.dedup();
         let nodes_len = nodes.len();
-        let addresses_and_connections_iter = nodes
-            .into_iter()
-            .map(|addr| (addr, connections.get(addr).cloned()));
+        let addresses_and_connections_iter = nodes.into_iter().map(|addr| async move {
+            let conn = match connections.get(addr).cloned() {
+                Some(conn) => Some(conn),
+                None => {
+                    // If it's a DNS endpoint, it could have been stored in the existing connections vector using the resolved IP address instead of the DNS endpoint's name.
+                    // We shall check if a connection is already exists under the resolved IP name.
+                    let (host, port) = match get_host_and_port_from_addr(addr) {
+                        Some((host, port)) => (host, port),
+                        None => return (addr, None),
+                    };
+                    match get_socket_addrs(host, port).await {
```

Comment:
maybe `ok` and `find` can simplify the code here
```
get_socket_addrs(host, port).await.ok()
  .map(|socket_addresses|socket_addresses.find(|addr|connections.get(&addr.to_string()).cloned())
```

Reply:
It would return the socket address, rather than the connection which I need